BL-736 TouchFree UI update

NatGreenwood-UL opened the pull request.

MattGrayUL committed:
Temporarily added ScreenControl UI to TouchFree project so the UI style matches

MattGrayUL committed:
removed some unneeded parts from ScreenControlUI import

MattGrayUL committed:
Apply the UI squish fix changes to the temporar TouchFree UI too for reference

MattGrayUL committed:
Removed unused 'TouchFreeRingCursor' script. Added 'SetCursorVisibility' to CursorManager & extended CursorManager to listen to 'V' key presses

MattGrayUL committed:
Merge branch 'develop' into feature/BL-737-Toggle_TouchFree_Cursor_Visbility

MattGrayUL committed:
Merge pull request #75 from ultraleap/bug/BL-749-TouchFree_UI_Settings_squish

bug/BL-749-TouchFree_UI_Settings_Squish_Fix

NatGreenwood-UL committed:
Intermediate TF UI update, add TF Build to gitignore

MattGrayUL committed:
Added third party colour picker and modified to make it branded. removed the use of TMPro

MattGrayUL committed:
Refactored CallToInteractController to better use the event and coroutine systems seen elsewhere.
* Prepared CallToInteractController for new config values

NatGreenwood-UL committed:
Move Fonts and Images into TouchFree folder in TF project

NatGreenwood-UL committed:
Add a ConfigManager & Redo the UI

NatGreenwood-UL committed:
Merge branch 'develop' into feature/BL-736-TouchFree_UI_Update

NatGreenwood-UL committed:
Add meta files for moved Fonts/Images

NatGreenwood-UL committed:
Resolve scene merge issues from develop

NatGreenwood-UL committed:
Resolve merge issues from develop

MattGrayUL committed:
Merge branch 'feature/BL-736-TouchFree_UI_Update' into feature/BL-742-Refactor_CTI_System

NatGreenwood-UL committed:
Add missing scene changes

MattGrayUL committed:
Merge branch 'feature/BL-736-TouchFree_UI_Update' into feature/BL-742-Refactor_CTI_System

MattGrayUL committed:
Added CTI links to the ConfigManager variables

MattGrayUL committed:
Merge branch 'feature/BL-742-Refactor_CTI_System' into feature/BL-737-Toggle_TouchFree_Cursor_Visbility

MattGrayUL committed:
Updated TouchFreeCursorManager to use ConfigManager values

MattGrayUL committed:
Fixed an issue where builds would error due to a private variable in CallToInteractController

MattGrayUL committed:
Changed case sensitivity in TransparentWIndow for CTICOntroller isShowing

MattGrayUL committed:
Changed cti active reference to live in TransparentWindow

MattGrayUL committed:
Merge branch 'feature/BL-742-Refactor_CTI_System' into feature/BL-737-Toggle_TouchFree_Cursor_Visbility

MattGrayUL committed:
Removed unneeded colour picker assets

NatGreenwood-UL committed:
Remove duplicated fonts/images

NatGreenwood-UL committed:
Relocate configManager & SliderInputField script, begin ConfigUI

NatGreenwood-UL committed:
Add missing files from last commit

MattGrayUL committed:
Merge branch 'feature/BL-736-TouchFree_UI_Update' into feature/BL-737-Toggle_TouchFree_Cursor_Visbility

# Conflicts:
#	TouchFree/Assets/TouchFree/Scripts/Configuration/ConfigManager.cs

MattGrayUL committed:
Added cursortype and cursorchanged event to cursormanager

MattGrayUL committed:
Merge branch 'feature/BL-736-TouchFree_UI_Update' into feature/BL-742-Refactor_CTI_System

# Conflicts:
#	TouchFree/Assets/TouchFree/Scripts/Configuration/ConfigManager.cs

MattGrayUL committed:
CallToInteractController updates from PR https://github.com/ultraleap/TouchFree/pull/82#discussion_r635135035 Removed already completed TODO, removed empty summary, fixed incorrect null check

MattGrayUL committed:
Updated ScreenContol Client version

MattGrayUL committed:
Change default CTI file path, added temporary getter to configmanager.config, Allowed CTIs to play.
* CTIController now reacts to the UI being deactivated when no hands are present
* Used static screenmanager events to avoid uninitialised scripts

MattGrayUL committed:
Added scene to build scripts

MattGrayUL committed:
Merge branch 'feature/BL-736-TouchFree_UI_Update' into experimental/BL-752-TouchFree_Colour_Picker

MattGrayUL committed:
reverted temporary 2s timer on ctis back to 10s

MattGrayUL committed:
Merge pull request #85 from ultraleap/experimental/BL-752-TouchFree_Colour_Picker

Experimental/bl 752 touch free colour picker

MattGrayUL committed:
Corrected spelling mistake for PR https://github.com/ultraleap/TouchFree/pull/82#discussion_r635329004

MattGrayUL committed:
Merge pull request #82 from ultraleap/feature/BL-742-Refactor_CTI_System

BL-742-Refactor_CTI_System

MattGrayUL committed:
Merge branch 'feature/BL-736-TouchFree_UI_Update' into feature/BL-737-Toggle_TouchFree_Cursor_Visbility

# Conflicts:
#	TouchFree/Assets/TouchFree/Scripts/Configuration/ConfigManager.cs

MattGrayUL committed:
TouchFreeCursorManager now overrides CursorManagers OnEnable&Disable

MattGrayUL committed:
Check for config on enable in TouchFreeCursorManager

MattGrayUL committed:
Merge pull request #83 from ultraleap/feature/BL-737-Toggle_TouchFree_Cursor_Visbility

Feature/bl 737 toggle touch free cursor visbility

NatGreenwood-UL committed:
Visuals mostly complete, logic for UI screen in process

MattGrayUL committed:
Fixed an issue where CTIs would still activate when a hand is present

MattGrayUL committed:
Fixed an issue where setting cursor visibility errors SetCursorVisibility is called before the execution order is managed

NatGreenwood-UL committed:
Add StandaloneFileBrowser to TouchFree

NatGreenwood-UL committed:
UI visuals complete, linked to configmanager

Pending work:
* Config files
* Button on home screen does not launch Service UI

NatGreenwood-UL committed:
Move home screen script & add logic to launch Service UI

NatGreenwood-UL committed:
Fix references to cursor previews and enable setting color of fillcursor

MattGrayUL committed:
Merge pull request #89 from ultraleap/feature/BL-737-Set_Cursor_Visibility

BL-737 Setting Cursor Visibility bug fix

MattGrayUL committed:
Merge pull request #90 from ultraleap/feature/BL-742-CTI_Refactor

BL-742 bug fix for CTIs appearing when they should not

NatGreenwood-UL committed:
Merge branch 'develop' into feature/BL-736-TouchFree_UI_Update

NatGreenwood-UL committed:
Connect cursors to configuration UI

MattGrayUL commented on .gitignore:
```diff
@@ -60,6 +60,7 @@ AssetDirStat.asset
 TouchFree_Build/
 Installer_Build/
+TouchFree_Build/
 
```
TouchFree_Build/ is duplicated

NatGreenwood-UL committed:
Remove ServiceUI holdover files

MattGrayUL commented on TouchFree/Assets/ScreenControl/ServiceUI/Scripts/Configuration/ConfigUI/ConfigUI.cs:
```diff
@@ -0,0 +1,82 @@
+            else if (_convertToStorageUnits)
+            {
+                //val = ScreenControlUtility.FromDisplayUnits(val);
+            }
```
do we need these anymore?

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/Configuration/ConfigManager.cs:
```diff
@@ -0,0 +1,77 @@
+                }
+
+                return _config;
+            }
```
be sure to add this when we add the config files

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/Configuration/ConfigManager.cs:
```diff
@@ -0,0 +1,77 @@
+        // CTI Settings
+        public bool ctiEnabled = true;
+        public string ctiFilePath = Path.Combine(Application.streamingAssetsPath, "CallTointeract/1 Push in mid-air to start.mp4");
+        public CtiHideTrigger ctiHideTrigger = CtiHideTrigger.PRESENCE;
```
This should default to INTERACTION

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/ScreenManager.cs:
```diff
@@ -0,0 +1,117 @@
+        if (Input.GetKeyDown(KeyCode.C))
+        {
+            SetUIActive(!isActive);
+        }
```
This could be removed

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/UI/HomeScreen.cs:
```diff
@@ -0,0 +1,106 @@
+{
+    public class HomeScreen : MonoBehaviour
+    {
+        [Space]
```
Not needed

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/UI/HomeScreen.cs:
```diff
@@ -0,0 +1,106 @@
+                {
+                    if (line.Contains("ScreenControl Service Version"))
+                    {
+                        version = line.Replace("ScreenControl Service Version: ", "");
```
This needs to be TouchFree?

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/UI/SliderInputFieldCombiner.cs:
```diff
@@ -0,0 +1,56 @@
+
+namespace Ultraleap.TouchFree
+{
+    // [DefaultExecutionOrder(-1)]
```
Remove comment?

MattGrayUL reviewed (commented):
Just a few comments while checking over changes so far.

Some scripts need namespaces but could be handled in the broader change to TouchFree if we wanted.

I haven't tested a build of this and believe we need to change the concept of the UI loading on every application launch before this can be merged in

NatGreenwood-UL committed:
TouchFree Config file

Note: The watcher to update the local copies when the file is changed is
not yet working

MattGrayUL committed:
Added configfilewatcher to scene. Added focus-based opening UI to ScreenManager. Removed UIManager as it is replaced by ScreenManager

MattGrayUL committed:
Changed configUI to set values without notifying the listeners

MattGrayUL committed:
Added default values to the touchlesscursor colors

NatGreenwood-UL committed:
Responses to code review comments

NatGreenwood-UL committed:
Fix some compile errors, and prevent the SetColorsToCorrectPreset
calling OnValueChanged

NatGreenwood-UL committed:
Changes to watch files less closely & load cursor colors on start

MattGrayUL committed:
Set cursor visibility via cursormanager on changes

MattGrayUL committed:
TouchFree UI appears by default on first ever launch of TouchFree. Subsequent launches require gaining of focus.

NatGreenwood-UL committed:
Connect CursorSize slider to cursor size (changed unit) and fix issue
where custom primary color was overwritten

NatGreenwood-UL committed:
Imported the latest Client from this same branch to ensure consistency
between TouchFree and Tooling

NatGreenwood-UL committed:
Bring the RingCursor size back to the same size as the FillCursor

MattGrayUL commented on ScreenControl_Unity/Assets/ScreenControl/Client/Scripts/Cursors/TouchlessCursor.cs:
```diff
@@ -18,23 +18,28 @@ public class TouchlessCursor : MonoBehaviour
+        // Variable: cursorDotSize
+        // The size of the dot when it isn't being shrunk
+        [SerializeField]
+        public float cursorDotSize = 0.25f;
```
Can we rename this to just 'cursorSize'? not all cursors will be dots

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/Configuration/ConfigFiles/ConfigFileWatcher.cs:
```diff
@@ -0,0 +1,52 @@
+{
+    public class ConfigFileWatcher : MonoBehaviour
+    {
+        private FileSystemWatcher touchfreeWatcher;
```
Not required

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/Configuration/ConfigFiles/ConfigFileWatcher.cs:
```diff
@@ -0,0 +1,52 @@
+
+        private void FileUpdated(object source, FileSystemEventArgs e)
+        {
+            Debug.Log("observed a file change in: " + e.Name);
```
Can remove this log

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/Cursors/TouchFreeCursorManager.cs:
```diff
@@ -1,25 +1,48 @@
+        void ConfigUpdated()
+        {
+            // Set cursorSize based on the slider
+
```
move comment to where it relates to?

MattGrayUL commented on TouchFree/Assets/TouchFree/Scripts/Configuration/ConfigManager.cs:
```diff
@@ -31,21 +43,23 @@ public class ConfigurationState
 
         // CTI Settings
         public bool ctiEnabled = true;
         public string ctiFilePath = Path.Combine(Application.streamingAssetsPath, "CallTointeract/1 Push in mid-air to start.mp4");
```
This path seems to not find a valid video

MattGrayUL reviewed (commented):
These comments cover the code review, I will also review a build and the project to ensure it is as expected

MattGrayUL committed:
Fixed an issue where cursors were visible by default on initialisation. Removed 'ringEnabled' from DotCursor.

NatGreenwood-UL committed:
Code Review Changes & Change version.txt location for TF Homepage

MattGrayUL committed:
Fixed a compile errror in dotcursor with the removed ringEnabled variable

NatGreenwood-UL committed:
ConfigUI now handles the hiding/showing of controls if enabled/disabled

Also UI will hide/show regions based on enabled/disabled toggles appropriately on startup now.

NatGreenwood-UL committed:
Explicitly call the Show/Hide based on toggle functions on boot
